### DISCUSSION TITLE
RaspberryMatic 3.75.7.20240420

### DISCUSSION THREAD

**Discussion by jens-maus** [OpenCCU/OpenCCU#2719] [Announcements]
This is release 3.75.7.20240420 of RaspberryMatic which is a **maintenance** release with the following bugfixes and feature changes:

[![Downloads](https://img.shields.io/github/downloads/jens-maus/RaspberryMatic/3.75.7.20240420/total.svg?style=flat-square)](https://github.com/jens-maus/RaspberryMatic/releases/3.75.7.20240420) ![License](https://img.shields.io/github/license/jens-maus/RaspberryMatic.svg?style=flat-square) [![Donate](https://img.shields.io/badge/donate-PayPal-green.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL) [![GitHub sponsors](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub&link=https://github.com/sponsors/jens-maus)](https://github.com/sponsors/jens-maus)

<sub>[Hier klicken für deutschsprachige 🇩🇪🇦🇹🇨🇭 Version des ChangeLogs/Diskussionsbeitrag](https://homematic-forum.de/forum/viewtopic.php?f=65&t=82334)</sub>

## 🚧 Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.75.6.20240316...3.75.7.20240420).</sub>

#### CCU/HomeMatic service changes:
- updated [OCCU](https://github.com/eq-3/occu) version to [3.75.7-1](https://github.com/jens-maus/occu/tree/b_3_75) integrating full compatibility to the [CCU3 3.75.7 firmware](https://homematic-ip.com/sites/default/files/other_media/ccu3-changelog-3-75-7.pdf) which comes with the following changes:
   - fixed error when determining the smallest/maximum permissible value when entering decimal numbers with decimal places.
- modify `S47InitRFHardware` init script to accept a `HMRF_ADDRESS` of `0x000000` and then generate a random RF address in these cases because `detect_radio_module` can return `0x000000` even if a RF module is BidCos/HmRF capable. (cf. https://homematic-forum.de/forum/viewtopic.php?f=65&t=82127).
- update integrated Mediola NEOserver to 2.15.0.

#### WebUI changes:
- modify `0119-WebUI-Fix-ProofAndSetValue` WebUI patch to comply with OCCU 3.75.7 changes.

#### Operating system changes:
- add `STOPSIGNAL SIGTERM` to default Dockerfile to explicitly make sure SIGTERM is used as a default stop signal if running in an OCI environment either under docker or podman (#2717).
- modify `S00watchdog` init script  so that the watchdog daemon will not be started in virtual OCI/LXC environments which could otherwise affect the watchdog environment of the corresponding host system (#2717).
- add kernel upstream patch to increase usb storgae quirks character limit to 256 characters so that the currently specified USB quirks string does not result in a kernel bootup error (cf. https://github.com/home-assistant/operating-system/issues/3308).
- update buildroot to 2024.02.1
- update rpi-eeprom for rpi4 to latest pieeprom-2024-04-15.bin.
- update rpi-eeprom to latest pieeprom-2024-04-05.bin for rpi5.
- update java azul to latest 11.72.19-ca-jre11.0.23
- update tailscale to 1.64.0
- update upstream kernel to 6.6.28
- update rpi-firmware to latest version.
- add u-boot patch for ODROID-N2 sd card maximum frequency limit to 40MHz.
- bump rpi kernel to use 6.6.x kernels and update to latest version (6.6.26).
- modify macvlan HA addon patch script to use the same cidr2network mechanism to identify the right subnet string as the `install-docker.sh` script wich should make use of this script slightly more comfort.
- update rpi3 platform device tree list to make sure all rpi3 related device trees are correcty deployed. (cf. https://github.com/home-assistant/operating-system/pull/3301).
- add buildroot upstream patch to introduce `BR2_LINUX_KERNEL_INSTALL_INTREE_OVERLAYS` to allow to install device tree overlays build with the kernel. Thus, adapted the rpi builds to use this new option and to completely refrain from using the `RPI_FIRMWARE` package for deploying the device tree overlay files.
- remove unnecessary `fixup.dat`/`start.elf` references in `config.txt` of rpi5 platform.
- enable multipath TCP (MPTCP) support (cf. https://github.com/home-assistant/operating-system/pull/3248).
- disable UAS via usb-storage.quirks on RPi for ADATA SD600Q (cf. https://github.com/home-assistant/operating-system/pull/3281).
- ignore warning in case of non-existing CP15 barrier instructions emulation (cf. https://github.com/home-assistant/operating-system/pull/3285).
- disable `CONFIG_WERROR` since buildroot disables it anyway.
- make sure `BR2_CCACHE_DIR` exists upon fresh execution.
- add new `rpi-eeprom-info` tool to output current RaspberryPi5 EEPROM/bootloader version and output if an update is available or not based on the `pieeprom.upd` file in `/boot`.
- rework `install-lxc.sh` script to perform slightly different default network bridge check than simply checking for the first line in `brctl`. (#2683).

## 👪 Contributors (alphabetically):
- @jens-maus

## 📝 Support:
For support on installation and help please visit the following web pages:

[RaspberryMatic - Documentation :us:](https://github.com/jens-maus/RaspberryMatic/wiki/en.Home)
[RaspberryMatic - Discussions :us:](https://github.com/jens-maus/RaspberryMatic/discussions)
[RaspberryMatic - Dokumentation :de:](https://github.com/jens-maus/RaspberryMatic/wiki)
[RaspberryMatic - Forum :de:](https://homematic-forum.de/forum/viewforum.php?f=65)

## 📦 Download:
The following installation archives can be downloaded for different hardware platforms. To verify their integrity a `sha256` checksum is provided as well. You can either upload these files using the WebUI-based update mechanism or unarchive them to e.g. flash the included `*.img` files on a fresh installation media (e.g. microSD card):

- RaspberryPi5 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-rpi5.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-rpi5.zip)<br/>SHA256: 8f4aac125d60d0f7338d62272916096c02106efeebbaf031e5748180b1731a35</sub>

- RaspberryPi4 Model B, RaspberryPi Compute Module 4, RaspberryPi 400 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-rpi4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-rpi4.zip)<br/>SHA256: 0151faf2a045c069a0180afe21030e601867e5b4cc27bf2558f6ad4815d758e1</sub>

- [CCU3](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3), [ELV-Charly](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-ELV-Charly), RaspberryPi3 Model B+, RaspberryPi3 Model B, RaspberryPi3 Model A+, RaspberryPi Compute Module 3, RaspberryPi Compute Module 3 lite, RaspberryPi Zero 2 W – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-rpi3.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-rpi3.zip)<br/>SHA256: d8b6ce28f28f645a755cde387bbdf8b4c53d03f854018801dccceecb9d731c6e</sub>
<sub>📦 [RaspberryMatic-3.75.7.20240420-ccu3.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-ccu3.tgz) (only for initial CCU3 Firmware -> RaspberryMatic Upgrade)<br/> SHA256: e48a8182982af42aaa63598e7b7be9ff8b7beba0188858e0faa68da54bd4d99d</sub>

- RaspberryPi2 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-rpi2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-rpi2.zip)<br/>SHA256: 868093a8ed193d1bca23a8d31bbf240aa72a6105475f79e4377bc65864d3cf48</sub>

- RaspberryPi Zero W, RaspberryPi Zero, RaspberryPi Compute Module 1, RaspberryPi1 (A+/B+) – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-rpi0.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-rpi0.zip)<br/>SHA256: d23f8c194fb57280c1872828e27f47da58bac069d77faf1b5da539f59f66ae92</sub>

- TinkerBoard S, TinkerBoard – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Tinkerboard)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-tinkerboard.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-tinkerboard.zip)<br/>SHA256: c4e9e924d4a91c96fc4a621f2ca44afa1717e31d862e90577f978372f10767de</sub>

- ODROID-N2/N2+/C4/C2 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Odroid)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-odroid-n2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-odroid-n2.zip)<br/>SHA256: 2ff1183d258781a858e5c56d57474732c55ec07fc7d25d5a52ddaf90b881762b</sub>
<sub>📦 [RaspberryMatic-3.75.7.20240420-odroid-c4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-odroid-c4.zip)<br/>SHA256: 3700cfae91ecc99c3b9f855c16a403c934fdb25d4019896cd538dfced7f16896</sub>
<sub>📦 [RaspberryMatic-3.75.7.20240420-odroid-c2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-odroid-c2.zip)<br/>SHA256: e8455c84c87eedd35d2d48f0016d496e8cf57accb1af309b150fcb19a7171af2</sub>

- Intel NUC – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-IntelNUC)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-intelnuc.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-intelnuc.zip)<br/>SHA256: 125419fe80e581bebd423f523ae7e4608370a475d338d95bfe52069a520715b9</sub>

- Open Virtual Appliance (OVA) – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE), [VirtualBox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-VirtualBox), [ESXi](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-ESXi), [Synology](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Synology-VMM), [QNAP](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QNAP-VirtualizationStation), [Workstation Player](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-Workstation-Player), [QEmu](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QEmu), [UNRAID](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-UNRAID), [HyperV](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HyperV)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-ova.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-ova.zip)<br/>SHA256: ec2a1b1ff86e844389cbf7d06ef6fbe74b9e8ac92a336b3b9f91bb55680910fe</sub>
<sub>📦 [RaspberryMatic-3.75.7.20240420.ova](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420.ova) (only for initial OVA installation)<br/>SHA256: 5848ad45bc2891fcaca8e903a3409b941a3961f712e891038a1709d6db15a83d</sub>

- Docker / Open Container Initiative (OCI) – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-oci_amd64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-oci_amd64.tgz) (amd64/x86_64)<br/>SHA256: e10de62b7e536a40cf42af8a31a8549da0c743637e6a26f67b03d0ee78a6a47f</sub>
<sub>📦 [RaspberryMatic-3.75.7.20240420-oci_arm64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-oci_arm64.tgz) (arm64/aarch64)<br/>SHA256: a6bd1f1b5f8e7e6c58e0411a03d2eb532ddd4bc8d7c944b3b931cf6db168d0c1</sub>
<sub>📦 [RaspberryMatic-3.75.7.20240420-oci_arm.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-oci_arm.tgz) (arm/armhf)<br/>SHA256: 534141e37964c65c356b823e88fd6030bfb7330705bbd21f2bedeb5b3f905e85</sub>

- LXC Container – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-LXC)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-lxc_amd64.tar.xz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-lxc_amd64.tar.xz) (amd64/x86_64)<br/>SHA256: 89125818b1f52832cb874d4b760df03b4e9dec980fe817d163949c460b0fa1da</sub>
<sub>📦 [RaspberryMatic-3.75.7.20240420-lxc_arm64.tar.xz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-lxc_arm64.tar.xz) (arm64/aarch64)<br/>SHA256: 6db3d4508f79e94c1680973087ca50c8deb2b20d30df3b5b2a90e46f01cab57b</sub>
<sub>📦 [RaspberryMatic-3.75.7.20240420-lxc_arm.tar.xz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-lxc_arm.tar.xz) (arm/armhf)<br/>SHA256: 50844fd2475f23a10120a69c8ed1f760fa564ac592b918cbd13d90f5d9e979a1</sub>

- Generic-aarch64 Appliance – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE)):
<sub>📦 [RaspberryMatic-3.75.7.20240420-generic-aarch64.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.75.7.20240420/RaspberryMatic-3.75.7.20240420-generic-aarch64.zip)<br/>SHA256: 7e7d1139747694607809ee610594d0cfe8a52e496c12bcb1a2b72f84245ff51e</sub>

- Kubernetes / K8s – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Kubernetes)</sub>

- Home Assistant Add-on – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant)</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/jens-maus/RaspberryMatic/releases/tag/3.75.7.20240420'>RaspberryMatic 3.75.7.20240420</a>.</em>

**Comment by RalfRoesler** [comment]
Hallo, nutze das Addon mit Home Assistant auf einem Mini-PC. Nach Update Einspielung ist es mir nicht mehr möglich auf Raspberrymatic zuzugreifen. Bad Gateway 502 Add-On scheint noch nicht bereit zusein, möglicherweise startet es noch.

Und nun?

**Reply by jens-maus** [reply]
Siehe hier:
https://github.com/jens-maus/RaspberryMatic/issues/2720

**Comment by Kleinhirn** [comment]
Hallo zusammen,
nach dem Update auf die "RaspberryMatic-3.75.7.20240420-tinkerboard.zip" Version, habe ich keinen Zugriff mehr auf meine Geräte. Die Weboberfläche ist anscheinend normal verfügbar, aber die Kommunikation mit dem Funkmodul scheint komplett gestört zu sein.
Nach einem Downgrade auf "RaspberryMatic-3.75.6.20240316-tinkerboard.zip", läuft wieder alles normal. Ein Backup wurde vorher erstellt.
LG

**Reply by Baxxy13** [reply]
Da ich kürzlich viele Tests mit einem Tinker-S gemacht hatte konnte ich das Problem auch sporadisch beobachten. 
Das Funkmodul wurde ab und zu nach einem Reboot einfach nicht erkannt. Ein weitere Reboot oder auch herunterfahren und kurz stromlos machen und dann ging's wieder. Die nächste Version sollte das Problem nicht mehr haben und auch allgemein deutlich runder laufen.
